rmgpy/rmg/model.py

```diff
@@ -1167,34 +1167,36 @@ def prune(self, reactionSystems, toleranceKeepInEdge, maximumEdgeSpecies, minSpe
         """
 
         ineligibleSpecies = []     # A list of the species which are not eligible for pruning, for any reason
-
-        numEdgeSpecies = len(self.edge.species)
+        prunableSpecies = reactionSystems[0].prunableSpecies
+        prunableNetworks = reactionSystems[0].prunableNetworks
+        
+        numPrunableSpecies = len(prunableSpecies)
         iteration = self.iterationNum
         # All edge species that have not existed for more than two enlarge
         # iterations are ineligible for pruning
-        for spec in self.edge.species:
+        for spec in prunableSpecies:
             if iteration - spec.creationIteration <= minSpeciesExistIterationsForPrune:
                 ineligibleSpecies.append(spec)
 
         # Get the maximum species rates (and network leak rates)
         # across all reaction systems
-        maxEdgeSpeciesRateRatios = numpy.zeros((numEdgeSpecies), numpy.float64)
+        maxEdgeSpeciesRateRatios = numpy.zeros((numPrunableSpecies), numpy.float64)
         for reactionSystem in reactionSystems:
-            for i in range(numEdgeSpecies):
+            for i in range(numPrunableSpecies):
                 rateRatio = reactionSystem.maxEdgeSpeciesRateRatios[i]
                 if maxEdgeSpeciesRateRatios[i] < rateRatio:
                     maxEdgeSpeciesRateRatios[i] = rateRatio
 
-            for i in range(len(self.networkList)):
-                network = self.networkList[i]
+            for i in range(len(prunableNetworks)):
+                network = prunableNetworks[i]
```

Comment:
I realize this wasn't really part of this PR, but could you replace this with `enumerate`? Sorry to keep adding more comments. I'll definitely merge after this.